nemo/core/classes/modelPT.py

```diff
@@ -1236,11 +1241,17 @@ def _is_restore_type_tarfile() -> bool:
         """
         global _MODEL_RESTORE_PATH
 
-        if _MODEL_RESTORE_PATH is None:
+        app_state = AppState()
+
+        if _MODEL_RESTORE_PATH is None and app_state.model_restore_path is None:
             return False
         else:
-            if tarfile.is_tarfile(_MODEL_RESTORE_PATH):
-                return True
+            if _MODEL_RESTORE_PATH:
+                if tarfile.is_tarfile(_MODEL_RESTORE_PATH):
+                    return True
+            elif app_state.model_restore_path:
+                if tarfile.is_tarfile(app_state.model_restore_path):
+                    return True
```

Comment:
This `_MODEL_RESTORE_PATH` ... It was a bad idea to make it a single string. I will need to refactor it to a list of strings in the future to support more than one model being restored in a sequence, and it will affect NLPModel restore_from impl. Will request your review at that time.